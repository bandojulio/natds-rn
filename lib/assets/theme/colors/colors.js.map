{"version":3,"sources":["colors.js"],"names":["colors","bw","black","gray","grayDark","grayDarker","grayLight","content","info","selectedBackground","feedback","error","primary","brown","orange","white","yellow","secondary","blue","green","purple"],"mappings":"yDAAO,GAAMA,uBAAS;AACpBC,GAAI;AACFC,MAAO,SADL;AAEFC,KAAM,SAFJ;AAGFC,SAAU,SAHR;AAIFC,WAAY,SAJV;AAKFC,UAAW,SALT,CADgB;;AAQpBC,QAAS;AACPC,KAAM,SADC;AAEPC,mBAAoB,WAFb,CARW;;AAYpBC,SAAU;AACRC,MAAO,SADC,CAZU;;AAepBC,QAAS;AACPC,MAAO,SADA;AAEPC,OAAQ,SAFD;AAGPC,MAAO,SAHA;AAIPC,OAAQ,SAJD,CAfW;;AAqBpBC,UAAW;AACTC,KAAM,SADG;AAETC,MAAO,SAFE;AAGTC,OAAQ,SAHC;AAITJ,OAAQ,SAJC,CArBS,CAAf","file":"colors.js","sourcesContent":["export const colors = {\n  bw: {\n    black: '#000000',\n    gray: '#BBBBBB',\n    grayDark: '#777777',\n    grayDarker: '#333333',\n    grayLight: '#EEEEEE',\n  },\n  content: {\n    info: '#227BBD',\n    selectedBackground: '#ff6b0b29',\n  },\n  feedback: {\n    error: '#E74627',\n  },\n  primary: {\n    brown: '#513528',\n    orange: '#FF6B0B',\n    white: '#FFFFFF',\n    yellow: '#F4AB34',\n  },\n  secondary: {\n    blue: '#71CACD',\n    green: '#B3CA36',\n    purple: '#7080A0',\n    yellow: '#FFDA1B',\n  },\n};\n"]}