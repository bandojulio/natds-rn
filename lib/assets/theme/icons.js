Object.defineProperty(exports,"__esModule",{value:true});exports.icons=undefined;var _natIcLeft=require('../icons/nat-ic-left.png');var _natIcLeft2=_interopRequireDefault(_natIcLeft);
var _natIcRight=require('../icons/nat-ic-right.png');var _natIcRight2=_interopRequireDefault(_natIcRight);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}

var iconSetup={
iconLeft:_natIcLeft2.default,
iconRight:_natIcRight2.default};


var icons=exports.icons=iconSetup;
//# sourceMappingURL=icons.js.map