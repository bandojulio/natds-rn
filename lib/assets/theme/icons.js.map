{"version":3,"sources":["icons.js"],"names":["iconSetup","iconLeft","natIcLeft","iconRight","natIcRight","icons"],"mappings":"iFAAA,mD;AACA,qD;;AAEA,GAAMA,WAAY;AAChBC,SAAUC,mBADM;AAEhBC,UAAWC,oBAFK,CAAlB;;;AAKO,GAAMC,qBAAQL,SAAd","file":"icons.js","sourcesContent":["import natIcLeft from '../icons/nat-ic-left.png';\nimport natIcRight from '../icons/nat-ic-right.png';\n\nconst iconSetup = {\n  iconLeft: natIcLeft,\n  iconRight: natIcRight,\n};\n\nexport const icons = iconSetup;\n"]}