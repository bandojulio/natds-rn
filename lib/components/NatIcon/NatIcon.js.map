{"version":3,"sources":["NatIcon.js"],"names":["NatIcon","name","size","fill","IconComponent","propTypes","PropTypes","string","number","defaultProps","ThemeNaturaLight","palette","text","primary"],"mappings":"wIAAA,4B;AACA,qC;AACA;AACA,+D;;AAEO,GAAMA,yBAAU,QAAVA,QAAU,MAA0B,IAAvBC,KAAuB,MAAvBA,IAAuB,CAAjBC,IAAiB,MAAjBA,IAAiB,CAAXC,IAAW,MAAXA,IAAW;AAC/C,GAAMC,eAAgB,cAAQH,IAAR,CAAtB;;AAEA;AACE,8BAAC,aAAD,EAAe,MAAOC,IAAtB,CAA4B,OAAQA,IAApC,CAA0C,MAAO,CAAEC,SAAF,CAAjD,iDADF;;AAGD,CANM;;AAQPH,QAAQK,SAAR,CAAoB;AAClBF,KAAMG,oBAAUC,MADE;AAElBN,KAAMK,oBAAUC,MAFE;AAGlBL,KAAMI,oBAAUE,MAHE,CAApB;;;AAMAR,QAAQS,YAAR,CAAuB;AACrBN,KAAMO,mCAAiBC,OAAjB,CAAyBC,IAAzB,CAA8BC,OADf;AAErBZ,KAAM,4BAFe;AAGrBC,KAAM,EAHe,CAAvB","file":"NatIcon.js","sourcesContent":["import React from 'react';\nimport PropTypes from 'prop-types';\nimport { load as loadSvg } from '../../assets/icons/svg';\nimport { ThemeNaturaLight } from '../../themes/ThemeNaturaLight';\n\nexport const NatIcon = ({ name, size, fill }) => {\n  const IconComponent = loadSvg(name);\n\n  return (\n    <IconComponent width={size} height={size} style={{ fill }} />\n  );\n};\n\nNatIcon.propTypes = {\n  fill: PropTypes.string,\n  name: PropTypes.string,\n  size: PropTypes.number,\n};\n\nNatIcon.defaultProps = {\n  fill: ThemeNaturaLight.palette.text.primary,\n  name: 'filled-brand-naturarosacea',\n  size: 24,\n};\n"]}