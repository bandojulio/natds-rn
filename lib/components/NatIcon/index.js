Object.defineProperty(exports,"__esModule",{value:true});var _NatIcon=require('./NatIcon');Object.defineProperty(exports,'NatIcon',{enumerable:true,get:function get(){return _NatIcon.NatIcon;}});
//# sourceMappingURL=index.js.map