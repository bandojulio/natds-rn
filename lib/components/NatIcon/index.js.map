{"version":3,"sources":["index.js"],"names":["NatIcon"],"mappings":"uLAASA,O","file":"index.js","sourcesContent":["export { NatIcon } from './NatIcon';\n"]}