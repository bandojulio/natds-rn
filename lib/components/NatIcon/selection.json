{
    "IcoMoonType": "selection",
    "icons": [
        {
            "icon": {
                "paths": [
                    "M512 973.197c254.016 0 460.403-206.388 460.403-460.403s-206.388-460.403-460.403-460.403-461.197 206.388-461.197 460.403 207.181 460.403 461.197 460.403zM512 115.895c218.295 0 396.105 177.811 396.105 396.105s-177.811 396.105-396.105 396.105-396.105-177.811-396.105-396.105 177.811-396.105 396.105-396.105z",
                    "M435.002 715.212c-18.257 0-32.546 14.288-32.546 32.546s14.288 32.546 32.546 32.546h213.532c18.257 0 32.546-14.288 32.546-32.546s-14.288-32.546-32.546-32.546h-74.617v-286.561c0-18.257-14.288-32.546-32.546-32.546h-106.369c-18.257 0-32.546 14.288-32.546 32.546s14.288 32.546 32.546 32.546h74.617v254.016h-74.617z",
                    "M550.102 289.736c0 28.935-23.456 52.391-52.391 52.391s-52.391-23.456-52.391-52.391c0-28.935 23.456-52.391 52.391-52.391s52.391 23.456 52.391 52.391z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "info"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 9,
                "id": 6,
                "name": "info",
                "prevSize": 32,
                "code": 59648
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 0
        },
        {
            "icon": {
                "paths": [
                    "M963.671 488.98l-428.651-428.651c-5.557-5.557-14.288-9.526-23.020-9.526s-17.464 3.969-23.020 9.526l-428.651 428.651c-12.701 12.701-12.701 33.34 0 46.040 6.35 6.35 14.288 9.526 23.020 9.526s16.67-3.175 23.020-9.526l373.085-373.085v778.716c0 18.257 14.288 32.546 32.546 32.546s32.546-14.288 32.546-32.546v-778.716l373.085 373.085c12.701 12.701 33.34 12.701 46.040 0s11.907-33.34 0-46.040z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "up-arrow"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 10,
                "id": 5,
                "name": "up-arrow",
                "prevSize": 32,
                "code": 59649
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 1
        },
        {
            "icon": {
                "paths": [
                    "M725.532 265.922h-427.064c-18.257 0-32.546 14.288-32.546 32.546s14.288 32.546 32.546 32.546h427.857c18.257 0 32.546-14.288 32.546-32.546-0.794-18.257-15.082-32.546-33.34-32.546z",
                    "M725.532 693.78h-427.064c-18.257 0-32.546 14.288-32.546 32.546s14.288 32.546 32.546 32.546h427.857c18.257 0 32.546-14.288 32.546-32.546-0.794-18.257-15.082-32.546-33.34-32.546z",
                    "M725.532 479.454h-427.064c-18.257 0-32.546 14.288-32.546 32.546s14.288 32.546 32.546 32.546h427.857c18.257 0 32.546-14.288 32.546-32.546-0.794-18.257-15.082-32.546-33.34-32.546z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "menu"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 4,
                "id": 4,
                "name": "menu",
                "prevSize": 32,
                "code": 59650
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 2
        },
        {
            "icon": {
                "paths": [
                    "M409.6 767.603c87.318 0 166.698-30.958 228.614-83.349l277.829 277.829c6.35 6.35 14.288 9.526 23.020 9.526s16.67-3.175 23.020-9.526c12.701-12.701 12.701-33.34 0-46.040l-277.829-277.829c51.597-61.916 83.349-142.090 83.349-228.614 0-197.656-160.347-358.003-358.003-358.003-196.862-0-358.003 161.141-358.003 358.003 0 197.656 161.141 358.003 358.003 358.003zM409.6 116.688c161.935 0 292.912 131.771 292.912 292.912 0 161.935-130.977 292.912-292.912 292.912s-292.912-131.771-292.912-292.912c-0-161.141 131.771-292.912 292.912-292.912z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "search"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 5,
                "id": 3,
                "name": "search",
                "prevSize": 32,
                "code": 59651
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 3
        },
        {
            "icon": {
                "paths": [
                    "M703.305 962.878c6.35 6.35 14.288 9.526 23.020 9.526s16.67-3.175 23.020-9.526c12.701-12.701 12.701-33.34 0-46.040l-404.837-404.837 404.837-404.837c12.701-12.701 12.701-33.34 0-46.040s-33.34-12.701-46.040 0l-428.651 427.857c-12.701 12.701-12.701 33.34 0 46.040l428.651 427.857z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "left-arrow"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 6,
                "id": 2,
                "name": "left-arrow",
                "prevSize": 32,
                "code": 59652
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 4
        },
        {
            "icon": {
                "paths": [
                    "M962.878 274.654c-12.701-12.701-33.34-12.701-46.040 0l-404.837 405.631-405.631-405.631c-12.701-12.701-33.34-12.701-46.040 0s-12.701 33.34 0 46.040l427.857 427.857c6.35 6.35 14.288 9.526 23.020 9.526 7.938 0 16.67-3.175 23.020-9.526l427.857-427.857c13.495-12.701 13.495-33.34 0.794-46.040z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "down-arrow"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 7,
                "id": 1,
                "name": "down-arrow",
                "prevSize": 32,
                "code": 59653
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 5
        },
        {
            "icon": {
                "paths": [
                    "M320.695 962.878c-6.35 6.35-14.288 9.526-23.020 9.526s-16.67-3.175-23.020-9.526c-12.701-12.701-12.701-33.34 0-46.040l404.837-404.837-404.837-404.837c-12.701-12.701-12.701-33.34 0-46.040s33.34-12.701 46.040 0l427.857 427.857c12.701 12.701 12.701 33.34 0 46.040l-427.857 427.857z"
                ],
                "attrs": [],
                "grid": 0,
                "tags": [
                    "right-arrow"
                ]
            },
            "attrs": [],
            "properties": {
                "order": 8,
                "id": 0,
                "name": "right-arrow",
                "prevSize": 32,
                "code": 59654
            },
            "setIdx": 0,
            "setId": 2,
            "iconIdx": 6
        }
    ],
    "height": 1024,
    "metadata": {
        "name": "icomoon"
    },
    "preferences": {
        "showGlyphs": true,
        "showQuickUse": true,
        "showQuickUse2": true,
        "showSVGs": true,
        "fontPref": {
            "prefix": "icon-",
            "metadata": {
                "fontFamily": "icomoon"
            },
            "metrics": {
                "emSize": 1024,
                "baseline": 6.25,
                "whitespace": 50
            },
            "embed": false,
            "autoHost": true
        },
        "imagePref": {
            "prefix": "icon-",
            "png": true,
            "useClassSelector": true,
            "color": 0,
            "bgColor": 16777215,
            "classSelector": ".icon",
            "name": "icomoon"
        },
        "historySize": 50,
        "showCodes": false,
        "gridSize": 16,
        "quickUsageToken": {
            "UntitledProject": "YTFhMmYxMzg1ZWMzNzVlNDhkY2ZhNzNjYjBjNjMwZGYjMSMxNTUzODc1NzU0IyMj"
        },
        "showLiga": true
    }
}