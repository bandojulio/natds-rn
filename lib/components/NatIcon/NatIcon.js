import React from 'react';
import PropTypes from 'prop-types';
import { createIconSetFromIcoMoon } from 'react-native-vector-icons';
import icoMoonConfig from './selection.json';
import { colors } from '../../assets/theme/NatColors';

const Icon = createIconSetFromIcoMoon(
  icoMoonConfig,
);

export const NatIcon = ({ name, size, color }) => (
  <Icon
    name={name}
    size={size}
    color={color}
  />
);

NatIcon.propTypes = {
  name: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
};

NatIcon.defaultProps = {
  size: 24,
  color: colors.bw.black,
};
