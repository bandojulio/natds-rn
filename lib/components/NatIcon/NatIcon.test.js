import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

import { NatIcon } from './NatIcon';
import { colors } from '../../assets/theme/colors';

describe('Icon component', () => {
  it('Should render icon component correctly', () => {
    const icon = renderer.create(
      <NatIcon name="nat_ic_info" />,
    ).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('Should render icon component with default styling', () => {
    const icon = renderer.create(
      <NatIcon name="nat_ic_info" />,
    ).toJSON();
    expect(icon).toHaveStyleRule('color', colors.bw.black);
    expect(icon).toHaveStyleRule('fontSize', 24);
    expect(icon).toHaveStyleRule('fontFamily', 'naturaiconography');
  });

  it('Should render icon component with custom styling', () => {
    const icon = renderer.create(
      <NatIcon name="nat_ic_info" color="red" size={100} />,
    ).toJSON();
    expect(icon).toHaveStyleRule('color', 'red');
    expect(icon).toHaveStyleRule('fontSize', 100);
    expect(icon).toHaveStyleRule('fontFamily', 'naturaiconography');
  });
});
