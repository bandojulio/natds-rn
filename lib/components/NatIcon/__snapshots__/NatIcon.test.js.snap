// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Should render icon component correctly 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 24,
      },
      undefined,
      Object {
        "fontFamily": "naturaiconography",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  î¤Š
</Text>
`;
